modules/shortcodes/tweet.php

```diff
@@ -46,33 +57,152 @@ public static function jetpack_tweet_shortcode( $atts ) {
 
 		self::$provider_args = $attr;
 
-		// figure out the tweet id for the requested tweet
-		// supporting both omitted attributes and tweet="tweet_id"
-		// and supporting both an id and a URL
+		/*
+		 * figure out the tweet id for the requested tweet
+		 * supporting both omitted attributes and tweet="tweet_id"
+		 * and supporting both an id and a URL
+		 */
 		if ( empty( $attr['tweet'] ) && ! empty( $atts[0] ) ) {
 			$attr['tweet'] = $atts[0];
 		}
 
 		if ( ctype_digit( $attr['tweet'] ) ) {
-			$id = 'https://twitter.com/jetpack/status/' . $attr['tweet'];
+			$id       = 'https://twitter.com/jetpack/status/' . $attr['tweet'];
```

Comment:
We need both the URLs and the ID here because WordPress.com uses the ID while Jetpack uses the full URL.